js/main.js

```diff
@@ -0,0 +1,174 @@
+'use strict';
+(function () {
+  var OFFER_VARIANTS = 8;
+  var MIN_POSITION_Y = 130;
+  var MAX_POSITION_Y = 630;
+  var RENT_VARIANTS = ['palace', 'flat', 'house', 'bungalo'];
+  var CHECKINS = ['12:00', '13:00', '14:00'];
+  var CHECKOUTS = ['12:00', '13:00', '14:00'];
+  var FEATURES = ['wifi', 'dishwasher', 'parking', 'washer', 'elevator', 'conditioner'];
+  var PHOTOS = [
+    'http://o0.github.io/assets/images/tokyo/hotel1.jpg',
+    'http://o0.github.io/assets/images/tokyo/hotel2.jpg',
+    'http://o0.github.io/assets/images/tokyo/hotel3.jpg'
+  ];
+  var fragment = document.createDocumentFragment();
+
+  var selectPinTemplate = document
+    .querySelector('#pin')
+    .content.querySelector('.map__pin');
+  var mapPins = document.querySelector('.map__pins');
+  var selectCardTemplate = document
+    .querySelector('#card')
+    .content.querySelector('.map__card');
+
+  function getRandomNumberOfOptionsFromList(options) {
+    var variants = [];
+
+    for (var i = 0; i < Math.floor(Math.random() * (options.length + 1)); i++) {
```

Comment:
Попробуй написать ручкой что происходит в каждой итерации. Если не заметишь, что вычисляется каждый раз, я тебе потом подскажу.

```suggestion
    for (var i = 0; i < Math.floor(Math.random() * options.length); i++) {
```
длинна массива у нас идет без +1, помнишь почему?